pgjdbc/src/main/java/org/postgresql/jdbc/ArrayDecoding.java

```diff
@@ -471,6 +475,12 @@ Object parseValue(String stringVal, BaseConnection connection) throws SQLExcepti
       throw org.postgresql.Driver.notImplemented(PgArray.class, "readArray(data,oid)");
     }
 
+    // 42.2.x should return enums as strings
+    int type = connection.getTypeInfo().getSQLType(typeName);
+    if (type == Types.CHAR || type == Types.VARCHAR) {
+      return (ArrayDecoder<A>) STRING_ONLY_DECODER;
+    }
```

Comment:
Is this relevant for 42.3?

Reply:
What's the plan for 42.3 ?

Reply:
enums as Strings seems a good mapping.
handling json and jsonb differently feels strange. But I would rather not require compatibility options to treat it the old way. So if we can make a clean break to handle json as a String (rather than PGObject), I think we should.

Reply:
We are at liberty to break things here. 
What do we return now for JSON ? Seems to me I'd rather return our own JSONObject? Do we know what others are doing here ?

Reply:
OK, an informal poll on twitter suggests String or InputReader. I would tend to agree.

Reply:
so i think we remove the comments at 378, 386, and 478 (specifying 42.2.x) and also uncomment the line at 387, to start treating json as String (rather than generic PGObject).

Reply:
done